homeworks/rostyslav.havuka_RostyslavHavuka/6-typeScript/src/index.ts

```diff
@@ -0,0 +1,247 @@
+        clearInterval(this.intervalId)
+        this.intervalId = setInterval(() => {
+          this.subScoreTimer(this.score)
+          this.sizeDown()
+          this.width = 1;
+          this.progressBar();
+        }, 3000);
+        this.gameState = true
+      }
+    })
+  }
+
+  private startKeysInterval(): any {
+    let possible: string = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz1234567890";
+    for (var i = 0; i < 1; i++)
+      this.currentKey = possible.charAt(Math.floor(Math.random() * possible.length));
+    keyElement.value = this.currentKey.toUpperCase()
+  }
+
+  private trueGame(): void {
+    if (this.gameState === true) {
+      this.startKeysInterval()
+      this.addScore()
+      this.setNewInterval()
+      this.sizeUp()
+      clearInterval(this.intervalStatusBarId)
+      this.width = 1;
+      this.timerProgressBar = 0;
+      this.progressBar()
+    } else {
+      alert('Please press "Start"')
+    }
+  }
+
+  private falseGame(): void {
+    if (this.gameState === true) {
+      this.subScore()
+      this.startKeysInterval()
+      this.setNewInterval()
+      this.sizeDown()
+      clearInterval(this.intervalStatusBarId)
+      this.width = 1;
+      this.timerProgressBar = 0;
+      this.progressBar()
+    } else {
+      alert('Please press "Start"')
+    }
+  }
+
+  private setKey(e: any): void {
+    if (e.key === keyElement.value.toLowerCase()) {
+      this.trueGame()
+    }
+    else {
+      this.falseGame()
+    }
+  }
+
+  private setNewInterval(): void {
+    if (this.gameState === true) {
+      clearInterval(this.intervalId)
+      this.intervalId = setInterval(() => {
+        this.subScoreTimer(this.score)
+        this.sizeDown()
+        clearInterval(this.intervalStatusBarId)
+        this.width = 1;
+        this.timerProgressBar = 0;
+        this.progressBar()
+      }, 3000);
+    }
+  }
+
+  private addScore(): void {
+    let rndom: number = this.randomNumber(5, 10)
+    this.score = this.score + rndom
+    cubeScoreElement.innerText = this.score.toString()
+    scoreElement.innerText = `+${rndom}`
+  }
+
+  private subScore(): void {
+    let rndom: number = this.randomNumber(20, 25)
+    this.score = this.score - rndom
+    cubeScoreElement.innerText = this.score.toString()
+    scoreElement.innerText = `-${rndom}`
+  }
+
+  private subScoreTimer(score: number): void {
+    let rndom: number = this.randomNumber(10, 15)
+    this.score = this.score - rndom
+    this.finishGame()
+    cubeScoreElement.innerText = this.score.toString()
+    scoreElement.innerText = `-${rndom}`
+  }
+
+  private subscribeOnKeyPress(): void {
+    window.addEventListener('keypress', (e) => {
+      this.setKey(e)
+      this.finishGame()
+    })
+  }
+
+  private randomNumber(min: number, max: number) {
+    return Math.floor(Math.random() * (max - min + 1)) + min;
+  }
+
+  private sizeUp(): void {
+    let obj: HTMLElement = document.getElementById("box");
```

Comment:
1. Don't use the id selector.

2. No external element dependencies. You already passed the box element through the constructor

3. You don't change the variable value. So assign it as `const`. Applicable to all variables below